doc/dev/adr/0023-property-support.md

```diff
@@ -0,0 +1,312 @@
+# ADR 23: SDK Support for Property
+
+## Context
+
+The AIO SDKs currently support two communication patterns: RPC and Telemetry.
+There is no support for the stateful communication pattern we call Property.
+There is an upcoming need to support this pattern because it is used heavily in OPC UA companion specifications, which AIO intends to support during [Br].
+
```

Comment:
Adding a high level comment - this pattern feels SO valuable, and it's already clear because we're using a _very_ similar pattern already in the ADR and DSS clients (that we do share a chunk of code for already as well!). So, a large goal that I would like to see for this pattern would be to make it generalizable enough to allow it to be used for these already defined scenarios, as well as (of course) the scenarios that this ADR was proposed for.